backend/common.go

```diff
@@ -90,6 +103,8 @@ type DataSourceInstanceSettings struct {
 
 	// JSONData contains the raw DataSourceConfig as JSON as stored by Grafana server. It repeats the properties in
 	// this object and includes custom properties.
+	//
+	// Deprecated: Use JSONDataMap instead.
```

Comment:
maybe we don't need to deprecate it. I guess it's okay to use JSONData outside of the `query` loop to parse it to a specific type like in https://github.com/grafana/grafana/blob/jsonData/pkg/tsdb/azuremonitor/azuremonitor.go#L89